framework/src/application/node/node.js

```diff
@@ -410,13 +416,11 @@ module.exports = class Node {
 			this.channel.publish('app:deleteBlock', eventData);
 
 			if (block.transactions.length) {
-				const transactions = block.transactions
-					.reverse()
-					.map(tx => this.chain.deserializeTransaction(tx));
-
-				transactions.forEach(transaction =>
-					this.transactionPool.add(transaction),
-				);
+				for (const transaction of block.transactions.reverse()) {
```

Comment:
```suggestion
				for (const transaction of block.transactions) {
```
No need to reverse anymore